test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js

```diff
@@ -429,24 +535,82 @@ define([
+                    setUpCellInState(this, 'INTERNAL_ERROR');
+                    await startRunningCell(this);
+
+                    expect(this.appCellWidgetInstance.__fsm().getCurrentState().state).toEqual(
+                        fsmState.INTERNAL_ERROR
+                    );
+                    // the reset button should be active
+                    const resetButton = this.kbaseNode.querySelector(selectors.reset);
+                    expect(resetButton).not.toHaveClass('hidden');
+                    expect(resetButton).not.toHaveClass('disabled');
+                });
+
+                it('shows an internal error if the FSM throws a hissy fit', async function () {
                     this.cell.metadata.kbase.appCell.fsm = {
-                        currentState: { mode: 'internal-error' },
+                        currentState: { this: 'does', not: 'exist' },
                     };
-                    await this.appCellWidgetInstance.init();
-                    await this.appCellWidgetInstance.attach(this.kbaseNode);
-                    await this.appCellWidgetInstance.start();
-                    await this.appCellWidgetInstance.run();
+                    await startRunningCell(this);
 
                     expect(this.appCellWidgetInstance.__fsm().getCurrentState().state).toEqual(
                         fsmState.INTERNAL_ERROR
                     );
-                    // the
+                    const expectedError = {
+                        title: 'Error initializing app state',
+                        message: 'Cannot find initial state',
+                        advice: [
+                            'Reset the app and try again.',
+                            'If that fails, delete the app cell and create a new one.',
+                        ],
+                        info: null,
+                        detail: null,
+                    };
+                    expect(this.appCellWidgetInstance.model.getItem('internalError')).toEqual(
+                        expectedError
+                    );
+
+                    [...expectedError.advice, expectedError.title, expectedError.message].forEach(
+                        (text) => {
+                            expect(this.kbaseNode.textContent).toMatch(text);
+                        }
+                    );
+                    // the reset button should be active
+                    const resetButton = this.kbaseNode.querySelector(selectors.reset);
+                    expect(resetButton).not.toHaveClass('hidden');
+                    expect(resetButton).not.toHaveClass('disabled');
+                });
+
+                Object.keys(indexedAppStates).forEach((state) => {
+                    // can't start in NEW -- it gets converted to an editing mode
+                    if (state !== 'NEW' && state.indexOf('BATCH') === -1) {
+                        it(`shows the correct execMessage when starting in state ${state}`, async function () {
```

Comment:
Nearly all of what the FSM module is currently doing in the app cell is setting the execMessage; the only other thing it does is to trigger the display of the results when a job completes. These tests are preparation for removing the FSM and using a simpler state system like that used in the BI cell -- the cell should start up with the same exec message regardless of what is happening behind the scenes.